src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.TrailingHeaders.cs

```diff
@@ -0,0 +1,225 @@
+                        Assert.DoesNotContain("amazingtrailer", data);
+                        Assert.DoesNotContain("Hello", data);
+                        Assert.DoesNotContain("World", data);
+                    }
+                }
+            });
+        }
+
+        [Fact]
+        public async Task GetAsyncResponseHeadersReadOption_TrailingHeaders_Available()
+        {
+            await LoopbackServer.CreateServerAsync(async (server, url) =>
+            {
+                using (HttpClientHandler handler = CreateHttpClientHandler())
+                using (var client = new HttpClient(handler))
+                {
+                    Task<HttpResponseMessage> getResponseTask = client.GetAsync(url, HttpCompletionOption.ResponseHeadersRead);
+                    await TestHelper.WhenAllCompletedOrAnyFailed(
+                        getResponseTask,
+                        server.AcceptConnectionSendCustomResponseAndCloseAsync(
+                            "HTTP/1.1 200 OK\r\n" +
+                            "Connection: close\r\n" +
+                            "Transfer-Encoding: chunked\r\n" +
+                            "Trailer: MyCoolTrailerHeader\r\n" +
+                            "\r\n" +
+                            "4\r\n" +
+                            "data\r\n" +
+                            "0\r\n" +
+                            "MyCoolTrailerHeader: amazingtrailer\r\n" +
+                            "Hello: World\r\n" +
+                            "\r\n"));
+
+                    using (HttpResponseMessage response = await getResponseTask)
+                    {
+                        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+                        Assert.Contains("chunked", response.Headers.GetValues("Transfer-Encoding"));
+                        Assert.Contains("MyCoolTrailerHeader", response.Headers.GetValues("Trailer"));
+
+                        // Pending read on the response content.
+                        Assert.Empty(response.TrailingHeaders);
+
+                        Stream stream = await response.Content.ReadAsStreamAsync();
+                        Byte[] data = new Byte[100];
+                        int start = 0;
+                        await stream.ReadAsync(data, start, 4);
+                        start = 4;
+
+                        // Intermedia test - haven't reached stream EOF yet.
+                        Assert.Empty(response.TrailingHeaders);
+                        Assert.Contains("data", System.Text.Encoding.Default.GetString(data));
+
+                        bool streamEOFReached = false;
+                        while (!streamEOFReached)
+                        {
+                            if (stream.Read(data, start, 10) == 0)
+                            {
+                                streamEOFReached = true;
+                            }
+                            start += 10;
```

Comment:
There's no need to keep track of start here, nor streamEOFReached… the loop can just be:
```C#
while (stream.Read(data, 0, data.Length) != 0);
```